web_m2x_options_manager/models/ir_model.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class IrModel(models.Model):
+    _inherit = "ir.model"
+
+    m2x_create_edit_option_ids = fields.One2many(
+        "m2x.create.edit.option",
+        "ir_model_id",
+    )
+
+    def button_empty(self):
+        for ir_model in self:
+            ir_model._empty_m2x_create_edit_option()
+
+    def button_fill(self):
+        for ir_model in self:
+            ir_model._fill_m2x_create_edit_option()
+
+    def _empty_m2x_create_edit_option(self):
+        """Removes every option for model ``self``"""
+        self.ensure_one()
+        self.m2x_create_edit_option_ids.unlink()
+
+    def _fill_m2x_create_edit_option(self):
+        """Adds every missing field option for model ``self``"""
+        self.ensure_one()
+        existing = self.m2x_create_edit_option_ids.mapped("ir_model_field_id")
+        valid = self.field_id.filtered(lambda f: f.ttype.startswith("many2"))
```

Comment:
```suggestion
        valid_ttypes = self.env["m2x.create.edit.option"]._get_valid_ttypes()
        valid = self.field_id.filtered(lambda f: f.ttype in valid_ttypes)
        existing = self.m2x_create_edit_option_ids.ir_model_field_id
```

Assuming you define `_get_valid_ttypes` in `m2x.create.edit.option` :)
I think it's better to have a specific method for this, and also to avoid matching only on startswith

Also note that in 14.0 you can drop the `mapped` call :)

Reply:
@ivantodorovich I think that adding method `_get_valid_ttypes` would be an overkill, these features are meant to be used for `Many2many` and `Many2one` fields only... I'd say it's pretty safe to leave these values here